### PR TITLE
adding forall, exists, lastOr to Process1

### PR DESCRIPTION
**Pull request by kaiserpelagic**

### CODE REVIEW

**Comment by pchiusano** [comment]
Hi thanks for this, but please make sure you merge with the `topic/redesign` branch before opening PRs. :)

The `NotNothing` parameter is no longer needed.

After you do that, follow the convention of `last` by defining a version as a `Pull`. There's a version in `pull1.scala` that doesn't output the last value, only returns it as a result. Then the version in `process1.last` is defined in terms of this. Any 'transducer' that just returns a single value should follow this same pattern.

**Comment by kaiserpelagic** [comment]
Whoops, sorry about not merging before pushing :)

I'll also include a version on Pull, a la last.

**Comment by pchiusano** [comment]
No prob.

**Comment by kaiserpelagic** [comment]
Wow, I really should have merged in upstream. Lots of good looking changes. `Stream[F,I] => Stream[F,Boolean]` instead of `Handle[F,I] => Pull[F,Boolean,Handle[F,I]]` is much more readable.

File: src/main/scala/fs2/Process1.scala [hunk 1]
```diff
@@ -31,6 +31,13 @@ object process1 {
   def drop[F[_],I](n: Long): Stream[F,I] => Stream[F,I] =
     _ pull (h => Pull.drop(n)(h) flatMap Pull.echo)
 
+  /**
+   * Emits a single `true` value if all input matches the predicate.
+   * Halts with `false` as soon as a non-matching element is received.
+   */
+  def forall[F[_], I](p: I => Boolean): Stream[F,I] => Stream[F,Boolean] =
```
Comment (pchiusano) [inline src/main/scala/fs2/Process1.scala]: Please keep methods in alphabetical order.

File: src/main/scala/fs2/pull1.scala [hunk 2]
```diff
@@ -75,6 +75,16 @@ private[fs2] trait pull1 {
   def echo[F[_],I]: Handle[F,I] => Pull[F,I,Nothing] =
     receive { case chunk #: h => Pull.output(chunk) >> echo(h) }
 
+  /** Write a single `true` value if all input matches the predicate, false otherwise*/
+  def forall[F[_],I](p: I => Boolean): Handle[F,I] => Pull[F,Nothing,Boolean] = {
```
Comment (pchiusano) [inline src/main/scala/fs2/pull1.scala]: alphabetical order please

**Comment by kaiserpelagic** [comment]
Added a few more Process1 functions to this PR. Hope you don't mind lumping them together.

**Comment by pchiusano** [comment]
Looks good. Combining them is totally fine.

**Pull request merged**